src/main/java/io/cryostat/net/web/http/api/v2/graph/PutActiveRecordingMetadataMutator.java

```diff
@@ -0,0 +1,154 @@
+import io.cryostat.recordings.RecordingTargetHelper;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+class PutActiveRecordingMetadataMutator
+        implements DataFetcher<HyperlinkedSerializableRecordingDescriptor> {
+
+    private final CredentialsManager credentialsManager;
+    private final TargetConnectionManager targetConnectionManager;
+    private final RecordingMetadataManager metadataManager;
+    private final RecordingTargetHelper recordingTargetHelper;
+    private final Provider<WebServer> webServer;
+    private final Gson gson;
+
+    @Inject
+    PutActiveRecordingMetadataMutator(
+            CredentialsManager credentialsManager,
+            TargetConnectionManager targetConnectionManager,
+            RecordingTargetHelper recordingTargetHelper,
+            RecordingMetadataManager metadataManager,
+            Provider<WebServer> webServer,
+            Gson gson) {
+        this.credentialsManager = credentialsManager;
+        this.targetConnectionManager = targetConnectionManager;
+        this.recordingTargetHelper = recordingTargetHelper;
+        this.metadataManager = metadataManager;
+        this.webServer = webServer;
+        this.gson = gson;
+    }
+
+    @Override
+    public HyperlinkedSerializableRecordingDescriptor get(DataFetchingEnvironment environment)
+            throws Exception {
+        GraphRecordingDescriptor source = environment.getSource();
+        ServiceRef target = source.target;
+        String uri = target.getServiceUri().toString();
+        String recordingName = source.getName();
+        Map<String, Object> settings = environment.getArgument("metadata");
+        Map<String, String> labels = new HashMap<>();
+
+        if (settings.containsKey("labels")) {
+            List<InputRecordingLabel> inputLabels =
+                    gson.fromJson(
+                            settings.get("labels").toString(),
+                            new TypeToken<List<InputRecordingLabel>>() {}.getType());
+            for (InputRecordingLabel l : inputLabels) {
+                labels.put(l.getKey(), l.getValue());
+            }
+        }
+
+        Metadata metadata =
+                metadataManager
+                        .setRecordingMetadata(uri, recordingName, new Metadata(labels))
+                        .get();
+
+        ConnectionDescriptor cd =
+                new ConnectionDescriptor(uri, credentialsManager.getCredentials(uri));
+
+        return targetConnectionManager.executeConnectedTask(
+                cd,
+                conn -> {
+                    IRecordingDescriptor desc =
+                            recordingTargetHelper.getDescriptorByName(conn, recordingName).get();
+
+                    WebServer ws = webServer.get();
+
+                    metadataManager.notifyRecordingMetadataUpdated(uri, recordingName, metadata);
```

Comment:
There can be a significant time delay between the `setRecordingMetadata` call and this `notify` occurring, since in the meantime we have to open up a remote JMX connection and query for recordings. Maybe the `setRecordingMetadata` call should be moved within this connected task?